benches/common.rs

```diff
@@ -0,0 +1,98 @@
+//! Common benchmark helpers shared by multiple benchmarks.
+
+// Not all helpers are used in all benchmarks.
+#![allow(dead_code)]
+
+use anyhow::Result;
+use std::path::PathBuf;
+use std::process::Command;
+use wasmtime::*;
+use wasmtime_wasi::{sync::WasiCtxBuilder, WasiCtx};
+
+pub fn build_wasi_example() {
+    println!("Building WASI example module...");
+    if !Command::new("cargo")
+        .args(&[
+            "build",
+            "--release",
+            "-p",
+            "example-wasi-wasm",
+            "--target",
+            "wasm32-wasi",
+        ])
+        .spawn()
+        .expect("failed to run cargo to build WASI example")
+        .wait()
+        .expect("failed to wait for cargo to build")
+        .success()
+    {
+        panic!("failed to build WASI example for target `wasm32-wasi`");
+    }
+
+    std::fs::copy(
+        "target/wasm32-wasi/release/wasi.wasm",
+        "benches/instantiation/wasi.wasm",
+    )
+    .expect("failed to copy WASI example module");
+}
+
+pub fn strategies() -> Vec<InstanceAllocationStrategy> {
+    vec![
+        // Skip the on-demand allocator when uffd is enabled
+        #[cfg(any(not(feature = "uffd"), not(target_os = "linux")))]
+        InstanceAllocationStrategy::OnDemand,
+        InstanceAllocationStrategy::pooling(),
+    ]
+}
+
+pub fn modules() -> Vec<&'static str> {
+    vec![
+        "empty.wat",
+        "small_memory.wat",
+        "data_segments.wat",
+        "wasi.wasm",
+    ]
+}
+
+pub fn make_engine(strategy: &InstanceAllocationStrategy, is_async: bool) -> Result<Engine> {
+    let mut config = Config::default();
+    config.allocation_strategy(strategy.clone());
+    config.async_support(is_async);
+    Engine::new(&config)
```

Comment:
Perhaps instead of a "make engine" function could this be a "make config" function? Dealing with boolean parameters typically requires commenting at each use-site what the parameter means and it's not all that hard to create an `Engine` from a config so dealing with just configs here might be easier?